pulp_container/app/models.py

```diff
@@ -256,6 +256,9 @@ class ContainerPushRepository(Repository, AutoAddObjPermsMixin, AutoDeleteObjPer
 
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
+        permissions = [
+            ("modify_content_containerrepository", "Can modify content in a repository"),
```

Comment:
Interesting, I'm not sure I understand then how it works.
We have the same permission defined in different classes? Does it work?

If we defined it once, why do we need it in 2 places? For auto-assignment I guess

Reply:
This does nothing for auto-assignment afaik.
I think it should be `modify_content_containerpushrepository`.

Reply:
It would not work until i would add this permission to this model for ContainerPushRepo.
We had this permission defined on ContainerRepo model but not on the Push repo model.

And yes, i needed to add it to the model because of auto-assignment otherwise i was observing this error:
```
(pulp) [vagrant@pulp3-source-fedora32 _scripts]$ pjournal 
-- Logs begin at Thu 2020-11-19 21:47:04 UTC. --
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:   File "/usr/local/lib/pulp/lib64/python3.8/site-packages/guardian/shortcuts.py", line 133, in assign_perm
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:     return model.objects.assign_perm(perm, user, obj)
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:   File "/usr/local/lib/pulp/lib64/python3.8/site-packages/guardian/managers.py", line 39, in assign_perm
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:     permission = Permission.objects.get(content_type=ctype, codename=perm)
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:   File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:     return getattr(self.get_queryset(), name)(*args, **kwargs)
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:   File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/query.py", line 406, in get
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]:     raise self.model.DoesNotExist(
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]: django.contrib.auth.models.Permission.DoesNotExist: Permission matching query does not exist.
Jan 28 20:07:38 pulp3-source-fedora32.fluffy.example.com gunicorn[488801]: 127.0.0.1 - - [28/Jan/2021:20:07:38 +0000] "POST /v2/test/this/blobs/uploads/ HTTP/1.1" 500 27 "-" "libpod/2.2.1"

```

Reply:
@mdellweg it works fine, user.has_perm on the object does give me true after i have just created the repo with that user

Reply:
after some investigation it turned out that we cannot create two permission with the same codename that would be used on different objects.